User inteface

eid-sonkrot opened the pull request.

Commit:
Implement Interactive User Interface Functions
In this commit, we lay the foundation for the project's interactive user interface by adding two crucial functions:
1.DisplayMenu: The DisplayMenu function presents users with a clear set of options, guiding them through various inventory management actions. These options include adding a product, viewing all products, editing a product, deleting a product, searching for a product, and exiting the system.
2. GetChoiceFromUser: The GetChoiceFromUser function ensures the user's choice is obtained correctly and validated. It prompts the user to enter a number representing their desired action. The input is then validated to ensure it falls within the valid range of 1 to 6, corresponding to the available menu options.

Alatrash814 commented on Inventory Management System/UserInterface.cs:
```diff
@@ -10,7 +10,70 @@ internal class UserInterface
+            {
+                Console.WriteLine("Welcome to the Inventory Management System!");
+                DisplayMenu();
+                int choice = GetChoiceFromUser();
```
don't use explicate variable types, please use "var", in case of null you can use, e.x: (int?) null;

Alatrash814 commented on Inventory Management System/UserInterface.cs:
```diff
@@ -10,7 +10,70 @@ internal class UserInterface
+                DisplayMenu();
+                int choice = GetChoiceFromUser();
+
+                switch (choice)
+                {
+                    case 1:
+                        //  add a product to the inventory
+                        break;
+                    case 2:
+                        // display all products in the inventory
+                        break;
+                    case 3:
+                        // edit an existing product in the inventory
+                        break;
+                    case 4:
+                        //  remove a product from the inventory
+                        break;
+                    case 5:
+                        //search for a product in the inventory
+                        break;
+                    case 6:
+                        Console.Clear();
+                        Console.WriteLine("Exiting the Inventory Management System. Goodbye!");
+                        System.Threading.Thread.Sleep(1000);
+                        return;
+                    
+                        
+                }
```
please move this to a separate method.

Alatrash814 commented on Inventory Management System/UserInterface.cs:
```diff
@@ -10,7 +10,70 @@ internal class UserInterface
     {
         static void Main(string[] args)
         {
+            
 
```
please remove the empty lines

Alatrash814 commented on Inventory Management System/UserInterface.cs:
```diff
@@ -10,7 +10,70 @@ internal class UserInterface
+                        Console.WriteLine("Exiting the Inventory Management System. Goodbye!");
+                        System.Threading.Thread.Sleep(1000);
+                        return;
+                    
+                        
```
please remove the empty lines

Alatrash814 commented on Inventory Management System/UserInterface.cs:
```diff
@@ -10,7 +10,70 @@ internal class UserInterface
+            }
+        }
+
+        static void DisplayMenu()
```
don't use static methods here, please convert the method to a non-static method.

Alatrash814 commented on Inventory Management System/UserInterface.cs:
```diff
@@ -10,7 +10,70 @@ internal class UserInterface
+            Console.WriteLine("6. Exit");
+        }
+
+        static int GetChoiceFromUser()
```
don't use static methods here, please convert the method to a non-static method.

Alatrash814 commented on Inventory Management System/UserInterface.cs:
```diff
@@ -10,7 +10,70 @@ internal class UserInterface
+        static int GetChoiceFromUser()
+        {
+            Console.Write("Enter your choice (1-6): ");
+            int choice;
```
```suggestion
            var choice = (int?) null;
```

Alatrash814 commented on Inventory Management System/UserInterface.cs:
```diff
@@ -10,7 +10,70 @@ internal class UserInterface
+                Console.WriteLine("Welcome to the Inventory Management System!");
+                DisplayMenu();
+                Console.Write("Invalid input. Enter a valid choice (1-6): ");
+
```
please remove the empty lines

Alatrash814 commented on Inventory Management System/UserInterface.cs:
```diff
@@ -10,7 +10,70 @@ internal class UserInterface
+            }
+            return choice;
         }
+
+
```
please remove the empty lines

Alatrash814 commented on Inventory Management System/UserInterface.cs:
```diff
@@ -10,7 +10,70 @@ internal class UserInterface
+
+
     }
+    
```
please remove the empty line

Alatrash814 commented on Inventory Management System/UserInterface.cs:
```diff
@@ -10,7 +10,70 @@ internal class UserInterface
+        {
+            Console.Write("Enter your choice (1-6): ");
+            int choice;
+            while (!int.TryParse(Console.ReadLine(), out choice) || choice < 1 || choice > 6)
+            {
+                Console.Clear();
+                Console.WriteLine("Welcome to the Inventory Management System!");
+                DisplayMenu();
+                Console.Write("Invalid input. Enter a valid choice (1-6): ");
```
```suggestion
            var isValidInput = int.TryParse(Console.ReadLine(), out choice);

            while (!isValidInput || choice < 1 || choice > 6)
            {
                Console.Clear();
                Console.WriteLine("Welcome to the Inventory Management System!");
                DisplayMenu();
                Console.Write("Invalid input. Enter a valid choice (1-6): ");
```

Alatrash814 commented:
Please change the class name from UserInterface.cs to InventoryManagementSystem.cs

Commit:
Refactor code to comply with the style guide

In this commit, we made several improvements to the code to adhere to the "cartera" style guide. The changes include:

1. Removing extra spaces for a cleaner code layout.
2. Converting all methods (except Main) to non-static methods for better object-oriented design.
3. Enhancing code readability by organizing the switch statement into a separate method and handling invalid input gracefully.

These changes improve the overall maintainability and readability of the codebase, making it easier for future development and collaboration.

Commit:
Refactor code to comply with the style guide "cartera"

In this commit, we made several improvements to the code to adhere to the "cartera" style guide. The changes include:

1. Removing extra spaces for a cleaner code layout.
2. Converting all methods (except Main) to non-static methods for better object-oriented design.
3. Enhancing code readability by organizing the switch statement into a separate method and handling invalid input gracefully.

These changes improve the overall maintainability and readability of the codebase, making it easier for future development and collaboration.

eid-sonkrot committed:
refactor code style

eid-sonkrot committed:
Refactor code Style

eid-sonkrot committed:
Merge branch 'master' into UserInteface
